components/search_bar/search_bar.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ChangeEvent, CSSProperties, FormEvent, useEffect, useRef} from 'react';
+
+import Constants from 'utils/constants';
+import * as Utils from 'utils/utils.jsx';
+import SearchSuggestionList from 'components/suggestion/search_suggestion_list.jsx';
+import SuggestionDate from 'components/suggestion/suggestion_date.jsx';
+import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
+import LoadingSpinner from 'components/widgets/loading/loading_spinner';
+import Provider from 'components/suggestion/provider';
+
+const {KeyCodes} = Constants;
+
+const style: Record<string, CSSProperties> = {
+    searchForm: {overflow: 'visible'},
+};
+
+type Props = {
+    searchTerms: string,
+    updateHighlightedSearchHint: (indexDelta: number, changedViaKeyPress?: boolean) => void,
+    handleChange: (e: ChangeEvent<HTMLInputElement>) => void,
+    handleSubmit: (e: FormEvent<HTMLFormElement>) => void,
+    handleEnterKey: (e: ChangeEvent<HTMLInputElement>) => void,
+    handleClear: () => void,
+    handleFocus: () => void,
+    handleBlur: () => void,
+    keepFocussed: boolean,
+    isFocussed: boolean,
+    suggestionProviders: Provider[],
+    isSearchingTerm: boolean,
+    isFocus: boolean,
+    isSideBarRight?: boolean,
+    getFocus?: (searchBarFocus: () => void) => void,
+    children?: React.ReactNode,
+}
+
+const defaultProps: Partial<Props> = {
+    isSideBarRight: false,
+    getFocus: (): void => {},
+    children: null,
+};
+
+const SearchBar: React.FunctionComponent<Props> = (props: Props): JSX.Element => {
+    const {isFocussed, keepFocussed, searchTerms, suggestionProviders} = props;
+
+    const searchRef = useRef<SuggestionBox>();
+
+    useEffect((): void => {
+        // let redux handle changes before focussing the input
+        if (isFocussed || keepFocussed) {
+            setTimeout(() => searchRef.current?.focus(), 0);
+        } else {
+            setTimeout(() => searchRef.current?.blur(), 0);
+        }
+    }, [searchTerms, isFocussed]);
+
+    const handleKeyDown = (e: ChangeEvent<HTMLInputElement>): void => {
+        if (Utils.isKeyPressed(e, KeyCodes.ESCAPE)) {
+            searchRef.current?.blur();
+            e.stopPropagation();
+            e.preventDefault();
+        }
+
+        if (Utils.isKeyPressed(e, KeyCodes.DOWN)) {
+            e.preventDefault();
+            props.updateHighlightedSearchHint(1, true);
+        }
+
+        if (Utils.isKeyPressed(e, KeyCodes.UP)) {
+            e.preventDefault();
+            props.updateHighlightedSearchHint(-1, true);
+        }
+
+        if (Utils.isKeyPressed(e, KeyCodes.ENTER)) {
+            props.handleEnterKey(e);
+        }
+    };
+
+    const getSearch = (node: SuggestionBox): void => {
```

Comment:
> note: doesn't need to be change as part of this PR but we need to file a separate ticket to get rid of this ref function or come up with another solution.

This is what the comment was referring to. This should be added to the redux state.